### PR TITLE
Fix bugs in the managed-nodegroups-go example

### LINKED ISSUE #577: Golang: library unusable: panic: reflect: Elem of invalid type interface {}
Library seem to be broken regarding Golang, including the basic example

https://github.com/pulumi/pulumi-eks/tree/master/examples/managed-nodegroups-go

## Expected behavior
to build 

## Current behavior

```
Previewing update (sam):
     Type                             Name                       Plan       Info
 +   pulumi:pulumi:Stack              managed-nodegroups-go-sam  create     1 error; 15 messages
 +   ├─ aws:iam:Role                  example-role0              create     
 +   ├─ aws:iam:Role                  example-role1              create     
 +   ├─ aws:iam:RolePolicyAttachment  example-role0-policy-2     create     
 +   ├─ aws:iam:RolePolicyAttachment  example-role0-policy-1     create     
 +   ├─ aws:iam:RolePolicyAttachment  example-role0-policy-0     create     
 +   └─ aws:iam:Role                  example-role2              create     
 
Diagnostics:
  pulumi:pulumi:Stack (managed-nodegroups-go-sam):
    panic: reflect: Elem of invalid type interface {}
    goroutine 68 [running]:
    reflect.(*rtype).Elem(0x10f77c0, 0xc0000c5d80, 0x194)
        /home/linuxbrew/.linuxbrew/Cellar/go/1.16.3_1/libexec/src/reflect/type.go:915 +0x1a5
    github.com/pulumi/pulumi/sdk/v3/go/pulumi.awaitInputs(0x14ba8b8, 0xc000130010, 0x11e7f80, 0xc00024f040, 0x16, 0x10f77c0, 0xc0000c5d80, 0x194, 0x0, 0x0, ...)
        /home/sam/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.0.0/go/pulumi/types.go:625 +0x2512
    github.com/pulumi/pulumi/sdk/v3/go/pulumi.awaitInputs(0x14ba8b8, 0xc000130010, 0x10f77c0, 0xc0000d0120, 0x194, 0x10f77c0, 0xc0000c5d80, 0x194, 0x0, 0x0, ...)
        /home/sam/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.0.0/go/pulumi/types.go:644 +0xc68
    github.com/pulumi/pulumi/sdk/v3/go/pulumi.awaitInputs(0x14ba8b8, 0xc000130010, 0x101d360, 0xc0000ce138, 0x97, 0x101d360, 0xc0000ce198, 0x197, 0x0, 0x0, ...)
        /home/sam/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.0.0/go/pulumi/types.go:680 +0x17a8
    github.com/pulumi/pulumi/sdk/v3/go/pulumi.toOutputWithContext.func1(0x101d360, 0xc0000ce138, 0x14cf148, 0xc0001d2230, 0x14de620, 0x101d360, 0x14ba8b8, 0xc000130010, 0x0)
        /home/sam/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.0.0/go/pulumi/types.go:753 +0x16e
    created by github.com/pulumi/pulumi/sdk/v3/go/pulumi.toOutputWithContext
        /home/sam/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.0.0/go/pulumi/types.go:745 +0x1eb
    exit status 2
 
    error: an unhandled error occurred: program exited with non-zero exit code: 1
```

## Steps to reproduce
trying to run https://github.com/pulumi/pulumi-eks/tree/master/examples/managed-nodegroups-go

## Context (Environment)
Bumping the version to v0.30.0 does not help

## Affected feature

Library is basically broken for Golang

### PR DESCRIPTION
**Pull request by lblackstone**
### Related issues (optional)
Fix #577

### CODE REVIEW

**Review by viveklak — APPROVED** [review]

**Comment by viveklak** [comment]
Should we roll a fix for https://github.com/pulumi/pulumi-eks/issues/521 into this? i.e. add the other go examples to https://github.com/pulumi/pulumi-eks/blob/master/examples/examples_go_test.go and re-enable go in the test matrix?

**Comment by lblackstone** [comment]
> Should we roll a fix for #521 into this? i.e. add the other go examples to https://github.com/pulumi/pulumi-eks/blob/master/examples/examples_go_test.go and re-enable go in the test matrix? I can follow up with fixes for the above in a separate PR if that is preferred.

Yes, we should enable the Go tests. I discovered that this change still isn't completely working, and didn't realize we weren't testing Go at all in CI.

File: examples/managed-nodegroups-go/main.go [hunk 1]
```diff
@@ -93,44 +94,46 @@ func main() {
...
-			_, err := eks.NewManagedNodeGroup(ctx,
-				"example-managed-ng1",
-				&eks.ManagedNodeGroupArgs{
+		result := pulumi.All(cluster.ToClusterOutput(), role1.ToRoleOutput(), role2.ToRoleOutput()).ApplyT(
+			func(args []interface{}) (string, error) {
+				cluster := args[0].(*eks.Cluster)
+				role1 := args[1].(iam.Role)
+				// Create a simple AWS managed node group using a cluster as input and the
+				// refactored API.
+				_, err := eks.NewManagedNodeGroup(ctx,
```
Comment (justinvp) [inline examples/managed-nodegroups-go/main.go:110]: We should fix up this example to avoid doing any creates inside an apply.

**Comment by lblackstone** [comment]
Rebased on #589

**Comment by lblackstone** [comment]
Let's re-enable the Go tests in a follow up PR.

I confirmed that this was working locally:
```
     Type                                   Name                                                             Status      Info
 +   pulumi:pulumi:Stack                    managed-nodegroups-go-eks-managed-ng                             created     1 message
 +   ├─ aws:iam:Role                        example-role0                                                    created
 +   ├─ aws:iam:Role                        example-role1                                                    created
 +   ├─ aws:iam:Role                        example-role2                                                    created
 +   ├─ aws:iam:RolePolicyAttachment        example-role1-policy-2                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role1-policy-0                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role1-policy-1                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role0-policy-1                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role0-policy-0                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role0-policy-2                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role2-policy-0                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role2-policy-2                                           created
 +   ├─ aws:iam:RolePolicyAttachment        example-role2-policy-1                                           created
 +   ├─ eks:index:Cluster                   example-managed-nodegroups-go                                    created
 +   │  ├─ eks:index:ServiceRole            example-managed-nodegroups-go-eksRole                            created
 +   │  │  ├─ aws:iam:Role                  example-managed-nodegroups-go-eksRole-role                       created
 +   │  │  ├─ aws:iam:RolePolicyAttachment  example-managed-nodegroups-go-eksRole-90eb1c99                   created
 +   │  │  └─ aws:iam:RolePolicyAttachment  example-managed-nodegroups-go-eksRole-4b490823                   created
 +   │  ├─ aws:ec2:SecurityGroup            example-managed-nodegroups-go-eksClusterSecurityGroup            created
 +   │  ├─ aws:eks:Cluster                  example-managed-nodegroups-go-eksCluster                         created
 +   │  ├─ aws:ec2:SecurityGroupRule        example-managed-nodegroups-go-eksClusterInternetEgressRule       created
 +   │  ├─ aws:ec2:SecurityGroup            example-managed-nodegroups-go-nodeSecurityGroup                  created
 +   │  ├─ aws:ec2:SecurityGroupRule        example-managed-nodegroups-go-eksExtApiServerClusterIngressRule  created
 +   │  ├─ aws:ec2:SecurityGroupRule        example-managed-nodegroups-go-eksClusterIngressRule              created
 +   │  ├─ aws:ec2:SecurityGroupRule        example-managed-nodegroups-go-eksNodeIngressRule                 created
 +   │  ├─ aws:ec2:SecurityGroupRule        example-managed-nodegroups-go-eksNodeClusterIngressRule          created
 +   │  ├─ aws:ec2:SecurityGroupRule        example-managed-nodegroups-go-eksNodeInternetEgressRule          created
 +   │  ├─ pulumi:providers:kubernetes      example-managed-nodegroups-go-eks-k8s                            created
 +   │  ├─ eks:index:VpcCni                 example-managed-nodegroups-go-vpc-cni                            created
 +   │  ├─ pulumi:providers:kubernetes      example-managed-nodegroups-go-provider                           created
 +   │  └─ kubernetes:core/v1:ConfigMap     example-managed-nodegroups-go-nodeAccess                         created
 +   ├─ eks:index:ManagedNodeGroup          example-managed-ng1                                              created
 +   │  └─ aws:eks:NodeGroup                example-managed-ng1                                              created
 +   ├─ eks:index:ManagedNodeGroup          aws-managed-ng0                                                  created
 +   │  └─ aws:eks:NodeGroup                aws-managed-ng0                                                  created
 +   └─ eks:index:ManagedNodeGroup          example-managed-ng2                                              created
 +      └─ aws:eks:NodeGroup                example-managed-ng2                                              created

Diagnostics:
  pulumi:pulumi:Stack (managed-nodegroups-go-eks-managed-ng):
    Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition

Outputs:
    kubeconfig: {
        apiVersion     : "v1"
        clusters       : [
            [0]: {
                cluster: {
                    certificate-authority-data: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM1ekNDQWMrZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRJeE1EVXlPREF6TXpneU9Gb1hEVE14TURVeU5qQXpNemd5T0Zvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBS2dnCjE5ZHNuK0V0NStUTStzL2J0bktIcWgxOUdEcXBCN2ZRL3UxUFJUVEx0ekIxQWpOQVFZWGYxa3pUdEtGNFdpTXUKcFhjalNoeXFTRmJJcjF2NUJqcTF4SDN4MWtwZ09GQU4yMWdxUk1qcE5mS1NiTXVPSEtwL2YwMlNEUGRTakRQVApwSnJsWCtuNDY0Vkc5R0ZZWmNoOGdUcWRDTzZaaHBpWWRHQmN6T01RR0dnL09wVlMvbVVqMXZSeE5IOUUrVzhICmtFMmlXczNMdmdtSVdUMlBiRGFIM0Y3cU1IVGFDRmkzTnpKLzcwanhnMXBOQld5NG5ZYmoveGR5bkErb2pGUTUKdklMMWp4ZitEWWdFR2lJczNPSkZDRytONkNQWmRpNXpyTE5vdzhVWEhaNmJMRmRpQTl1NlRjSU11Qk02US9PZgppcVhkRXRJN1FNZE5zWDgyUWlrQ0F3RUFBYU5DTUVBd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0hRWURWUjBPQkJZRUZGSVZ1OWNIaHR1Njl2UVlGSnkxZUlKK29QdnJNQTBHQ1NxR1NJYjMKRFFFQkN3VUFBNElCQVFCMDFzYzBpRzB0ZG41ODNUUE8reWtQdElnS3lvM29jWCtaM0ticERNSEkvR1poZUo0aQprRWVHUVl3RU5FUEpHSUJyNGs0RFlXYzgrU1VDNVZwQ0tJZGp2ZTdrWFl2RUVJdi92R3R1ajMzcXFCdDN4ZWFyCjFoZzhvVm9ZWFZVOWdyMW1ESGUxMjFFSmo4YWk0T1FDcVNkNHRiSGVzYllOanhaRE52L2pIQ3QramhkaWg4OEMKZllmcEw2M0x1TlZYbWhIazB6dDhQcE1LaG54cm9qcEVVZC9FZEl3V21lUk5wSnliaWhubit1a3FjbVVUd0V1bgpBZ1lQVCsreUtnT080clUrWjVxc0x2Tms0UGNYTGt6T3c3MnFXTzRVdHZySExBRkJESml5K3FsbmJ5VWRtR2NMCmQ5b3BhekRERlZEUUpMZHFYeUR5T1NRT3hCbVJKazVVL3lmSAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg=="
                    server                    : "https://4F86C6874E0210A378CF3A057C415D7A.gr7.us-west-2.eks.amazonaws.com"
                }
                name   : "kubernetes"
            }
        ]
        contexts       : [
            [0]: {
                context: {
                    cluster: "kubernetes"
                    user   : "aws"
                }
                name   : "aws"
            }
        ]
        current-context: "aws"
        kind           : "Config"
        users          : [
            [0]: {
                name: "aws"
                user: {
                    exec: {
                        apiVersion: "client.authentication.k8s.io/v1alpha1"
                        args      : [
                            [0]: "eks"
                            [1]: "get-token"
                            [2]: "--cluster-name"
                            [3]: "example-managed-nodegroups-go-eksCluster-b978653"
                        ]
                        command   : "aws"
                        env       : [
                            [0]: {
                                name : "AWS_PROFILE"
                                value: "pulumi-dev-sandbox"
                            }
                        ]
                    }
                }
            }
        ]
    }

Resources:
    + 37 created

Duration: 21m18s
```

**Review by justinvp — APPROVED** [review]

**Pull request merged**